src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -543,6 +577,11 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
                 (fdv_end, market_cap_end) =
                     fdv_market_cap(reserves_end, supply_minuend_end);
 
+                // Refund market registration deposit.
+                let RegistrantDeposit { market_registrant, deposit } =
+                    move_from<RegistrantDeposit>(market_address);
+                coin::deposit(market_registrant, deposit);
```

Comment:
There's no way this can get blocked, right? In case of a frozen coin store (I can't remember if you can even freeze APT 😅) or some weird edge case? Only asking because if so it'd lock a market out of ever being able to exit the bonding curve

Maybe `aptos_account::transfer` just in case?

I know that by virtue of having registered a market they must have APT, but just figure might as well make this as flexible as possible

Reply:
> can get blocked
> ...
> if so it'd lock a market out of ever being able to exit the bonding curve

First of all, I appreciate the intuition on this. Thanks for raising the issue.

> [`aptos_account::transfer`]

That API expects APT at the `source` account, but the refundable deposit is
instead held as a `store`-able field in the new `RegistrantDeposit` so as to
avoid it being accounted for in pool calculations.

Alternatively there is [`aptos_account::deposit_coins`], which performs a check
on [`aptos_account::can_receive_direct_coin_transfers`]. Notably, in the general
case a user could disable the ability to receive direct coin transfers, e.g.
receive coins they have not explicitly registered, via
[`aptos_account::set_allow_direct_coin_transfers`]. So for example there could
be a problem if the refundable deposit was made in `APT` but paid out in
emojicoin, for a registrant who never traded and who turned off direct coin
transfers.

However, in our case it is `APT` in both directions and the registrant
necessarily has to have registered an `AptosCoin`-type `CoinStore` in order to
pay the original market registration deposit (or more specifically, to have had
a proxy for one via FA migration).  Note too that internally,
[`aptos_account::deposit_coins`] calls on `coin::deposit` anyways. So I don't
think that changing APIs would have any functional effect.

> can even freeze APT

This is the real question, and importantly
[`aptos_coin::initialize` destroys the feeze capability].

[`aptos_coin::initialize` destroys the feeze capability]: https://github.com/aptos-labs/aptos-core/blob/f6e43ee347a012ebcbdd7c36933fa99aafeba956/aptos-move/framework/aptos-framework/sources/aptos_coin.move#L54
[`aptos_account::set_allow_direct_coin_transfers`]: https://github.com/aptos-labs/aptos-core/blob/f6e43ee347a012ebcbdd7c36933fa99aafeba956/aptos-move/framework/aptos-framework/sources/aptos_account.move#L145
[`aptos_account::can_receive_direct_coin_transfers`]: https://github.com/aptos-labs/aptos-core/blob/f6e43ee347a012ebcbdd7c36933fa99aafeba956/aptos-move/framework/aptos-framework/sources/aptos_account.move#L182
[`aptos_account::transfer`]: https://github.com/aptos-labs/aptos-core/blob/f6e43ee347a012ebcbdd7c36933fa99aafeba956/aptos-move/framework/aptos-framework/sources/aptos_account.move#L76
[`aptos_account::deposit_coins`]: https://github.com/aptos-labs/aptos-core/blob/f6e43ee347a012ebcbdd7c36933fa99aafeba956/aptos-move/framework/aptos-framework/sources/aptos_account.move#L116

Reply:
Gotcha, awesome thanks for the insight, I thought that they removed the freeze but things have changed and wasn't sure if I'd just missed some crazy change... 🤣